packages/python/accern_data/accern_data.py

```diff
@@ -292,56 +354,88 @@ def __init__(
             token: str) -> None:
         self._base_url = url
         self._token = token
-        self._filters = self.validate_filters(None)
+        self._filters = self.parse_filters(self.validate_filters(None))
         self._params: Dict[str, str] = {}
         self._mode: Optional[Mode] = None
         self._first_error = True
         self._expected_records: List[int] = []
 
     @staticmethod
-    def validate_filters(filters: Optional[FiltersType]) -> FiltersType:
+    def validate_filters(
+            filters: Optional[FiltersType]) -> Dict[
+                str, Optional[Union[bool, int, str]]]:
         if filters is None:
             return {}
-        for key in filters.keys():
-            if key not in FilterField:
+        valid_filters: Dict[str, Optional[Union[bool, int, str]]] = {}
+        for key, value in filters.items():
+            if key not in FILTER_FIELD:
                 raise ValueError(
-                    f"{key} is not a valid filed."
-                    f"Possible fileds: {FilterField}")
-        return filters
+                    f"{key} is not a valid field."
+                    f"Possible fields: {FILTER_FIELD}")
+            assert isinstance(value, (bool, int, str)) or value is None
+            valid_filters[key] = value
+        return valid_filters
+
+    @staticmethod
+    def parse_filters(
+            filters: Dict[str, Optional[Union[bool, int, str]]]) -> Dict[
+                str, Optional[str]]:
+        proper_filters: Dict[str, Optional[str]] = {}
+        for key, value in filters.items():
+            if isinstance(value, bool):
+                proper_filters[key] = f"{value}".lower()
+            elif value is None:
+                proper_filters[key] = value
+            else:
+                proper_filters[key] = f"{value}"
+        return proper_filters
 
     def set_filters(self, filters: FiltersType) -> None:
-        self._filters = self.validate_filters(filters)
+        self.set_raw_filters(self.validate_filters(filters))
 
-    def set_mode(self, mode: MODE) -> None:
-        if mode not in ALL_MODES:
-            raise ValueError(
-                f"Please set proper mode. It is '{mode}' which is not in "
-                f"{ALL_MODES}")
+    def set_raw_filters(
+            self, filters: Dict[str, Optional[Union[bool, int, str]]]) -> None:
+        for key in filters.items():
```

Comment:
not items

`assert not (filters.keys() & set(EXCLUDED_FILTER_FIELD))`

shouldn't this be in parse_filters?